skrub/_expressions/_skrub_namespace.py

```diff
@@ -746,20 +897,83 @@ def eval(self, environment=None):
                 "The `environment` passed to `eval()` should be None or a dictionary, "
                 f"got: '{type(environment)}'"
             )
+        if environment is None and (
+            keep_subsampling or not uses_subsampling(self._expr)
+        ):
+            # In this configuration the result is the same as the preview so
+            # we call preview() to benefit from the cached result.
+
+            # Before returning, we trigger an error if keep_subsampling=True was
+            # passed but no subsampling was configured:
+            _ = env_with_subsampling(self._expr, {}, keep_subsampling)
+            return self.preview()
         if environment is None:
-            mode = "preview"
-            clear = False
+            environment = self.get_data()
         else:
-            mode = "fit_transform"
-            clear = True
             environment = {
                 **environment,
                 "_skrub_use_var_values": not _var_values_provided(
                     self._expr, environment
                 ),
             }
+        environment = env_with_subsampling(self._expr, environment, keep_subsampling)
+        return evaluate(
+            self._expr, mode="fit_transform", environment=environment, clear=True
+        )
+
+    def preview(self):
+        """Get the value computed for previews (shown when printing the expression).
+
+        Returns
+        -------
+        preview result
+            The result of evaluating the expression on the data stored in its
+            variables.
+
+        See Also
+        --------
+        Expr.skb.subsample_previews :
+            Specify how to subsample an intermediate result when computing
+            previews.
```

Comment:
OK, so what I seem to understand is that "subsample_previews" sets the subsample for the whole subsequent operations, unlike "preview"? Is that right?

I'd love to find a better name. I don't find that when reading code it's terribly clear for me what is happening.

Reply:
`preview()` returns the actual computed value of the preview (for any node, whether there is subsampling or not)

`subsample_previews()` adds a node in the graph that performs subsampling (or not, when `keep_subsampling=False`).

I tried to make the distinction with the plural but I guess that's not clear. maybe it should be just `subsample()` -- I just wanted to hint in the name that by default subsampling only takes place for computing previews.